src/parser/paladin/holy/modules/SelfHealing.js

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import Events from 'parser/core/Events';
+import { formatThousands, formatPercentage } from 'common/format';
+import HealingDone from 'parser/shared/modules/HealingDone';
+
+import { DIRECT_SELF_HEALING_ABILITIES } from '../constants';
+
+/*
+ * Calculates targeted healing on self
+ */
+
+class SelfHealing extends Analyzer {
+    static dependencies = {
+        healingDone: HealingDone,
+    };
+
+    _healing = 0;
+    _overhealing = 0;
+    _selfHealPercent = 0;
+
+    constructor(...args) {
+        super(...args);
+        this.addEventListener(Events.heal.by(SELECTED_PLAYER), this._onCast);
```

Comment:
You can leverage some of the `addEventListener` magic:
- you can make it accept only the spells that are in `DIRECT_SELF_HEALING_ABILITIES`
    - to do that, make it an array of spell *objects* (omit the `id`)
    ```javascript
    export const DIRECT_SELF_HEALING_ABILITIES = [
      SPELLS.HOLY_SHOCK_HEAL,
      SPELLS.HOLY_LIGHT,
      SPELLS.FLASH_OF_LIGHT,
      SPELLS.BESTOW_FAITH_TALENT,
    ];
    ```
    - use `spell()` method with the array
    ```javascript
    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(DIRECT_SELF_HEALING_ABILITIES), this._onCast);
    ```
- you can also simplify the target check by using `to()` method

- all together, you can use this:
```javascript
this.addEventListener(Events.heal.by(SELECTED_PLAYER).to(SELECTED_PLAYER).spell(DIRECT_SELF_HEALING_ABILITIES), this._onCast);
```

(also, change `_onCast` to `_onHeal` as you mentioned and get rid of both conditions on lines 32 and 33, it wil lbe taken care of)